src/GraphQL/Execution/ExecutionHelper.cs

```diff
@@ -143,8 +145,18 @@ public static void AssertValidValue(ISchema schema, IGraphType type, object inpu
 
             if (type is ScalarGraphType scalar)
             {
-                if (ValueFromScalar(scalar, input) == null)
-                    throw new InvalidValueException(fieldName, "Invalid Scalar value for input field.");
+                // verify value can be converted successfully
+
+                if (input is IValue value)
+                {
+                    if (scalar.ParseLiteral(value) == null)
+                        throw new InvalidValueException(variableName, $"Unable to convert '{value.Value}' to {type.Name}");
```

Comment:
Below I can see messages
` $"Unable to parse input as a '{type.Name}' type. Did you provide a List or Scalar value accidentally?");`
and
`   $"Unrecognized input fields {string.Join(", ", unknownFields.Select(k => $"'{k}'"))} for type '{type.Name}'.");`

with quoted `type.Name`. Let's do here and below also.

Reply:
```suggestion
                        throw new InvalidValueException(variableName, $"Unable to convert '{value.Value}' to '{type.Name}'");
```